src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -76,6 +76,24 @@ public override string Description
         /// </summary>
         protected object Expected { get; }
 
+        /// <summary>
+        /// Returns a new DictionaryContainsKeyValuePairConstraint checking for the
+        /// presence of a particular key-value-pair in the dictionary.
+        /// </summary>
+        public DictionaryContainsKeyValuePairConstraint WithValue(object expected)
+        {
+            var builder = this.Builder;
+            if (builder == null)
+            {
+                builder = new ConstraintBuilder();
+                builder.Append(this);
+            }
+
+            var constraint = new DictionaryContainsKeyValuePairConstraint(Expected, expected);
+            builder.Append(constraint);
+            return constraint;
```

Comment:
I was a bit unsure about this part, but it seems to handle `Assert.That(dictionary, Does.Not.ContainKey(...).WithValue....` nicely

Reply:
Also not sure what's the best way to solve it since I'm not too deeply into the code base. I found that approach in another class (not sure anymore where it was). So if there's a better way, let me know!